File: packages/generators/ruby/seed/model/trace/lib/SeedClient/v_2/problem/types/file_info_v_2.rb
# frozen_string_literal: true
require "json"

module SeedClient
  module V2
    module Problem
      class FileInfoV2
        attr_reader :filename, :directory, :contents, :editable, :additional_properties
        # @param filename [String] 
        # @param directory [String] 
        # @param contents [String] 
        # @param editable [Boolean] 
        # @param additional_properties [OpenStruct] Additional properties unmapped to the current class definition
        # @return [V2::Problem::FileInfoV2] 
        def initialze(filename:, directory:, contents:, editable:, additional_properties: nil)
          # @type [String] 
          @filename = filename
          # @type [String] 
          @directory = directory
          # @type [String] 
          @contents = contents
          # @type [Boolean] 
          @editable = editable
          # @type [OpenStruct] 
          @additional_properties = additional_properties
        end
        # Deserialize a JSON object to an instance of FileInfoV2
        #
        # @param json_object [JSON] 
        # @return [V2::Problem::FileInfoV2] 
        def self.from_json(json_object:)
          struct = JSON.parse(json_object, object_class: OpenStruct)
          filename = struct.filename
          directory = struct.directory
          contents = struct.contents
          editable = struct.editable
          new(filename: filename, directory: directory, contents: contents, editable: editable, additional_properties: struct)
        end
        # Serialize an instance of FileInfoV2 to a JSON object
        #
        # @return [JSON] 
        def to_json
          {
 filename: @filename,
 directory: @directory,
 contents: @contents,
 editable: @editable
}.to_json()
        end
      end
    end
  end
end

File: seed/php-model/trace/src/V2/V3/Problem/FileInfoV2.php
<?php

namespace Seed\V2\V3\Problem;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class FileInfoV2 extends JsonSerializableType
{
    /**
     * @var string $filename
     */
    #[JsonProperty('filename')]
    public string $filename;

    /**
     * @var string $directory
     */
    #[JsonProperty('directory')]
    public string $directory;

    /**
     * @var string $contents
     */
    #[JsonProperty('contents')]
    public string $contents;

    /**
     * @var bool $editable
     */
    #[JsonProperty('editable')]
    public bool $editable;

    /**
     * @param array{
     *   filename: string,
     *   directory: string,
     *   contents: string,
     *   editable: bool,
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->filename = $values['filename'];$this->directory = $values['directory'];$this->contents = $values['contents'];$this->editable = $values['editable'];
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}
